app/models/hmpps_api/offender_base.rb

```diff
@@ -217,8 +217,16 @@ def within_early_allocation_window?
       earliest_date.present? && earliest_date <= Time.zone.today + 18.months
     end
 
+    def needs_com_and_ldu_is_uncontactable?
+      needs_com? && sentenced? && allocated_com_name.nil? && ldu_email_address.nil?
```

Comment:
## ⚠️ Potential bug alert 🐛 

We might want to use `.empty?` instead of `.nil?` here to avoid accidentally letting empty strings through – for COM name and LDU email address.

(Although I realise that the new LDU model does require a valid email address... so maybe this is a moot point.)